### PR TITLE
feat: integrate wsc virtual room field on calendar module 

### PR DESCRIPTION
**Pull request by lisaparma**
This PR modifies the logic that handles the selection of a virtual room based on the presence of integrations with the Chats and Workstream Collaboration modules.
	•	If only Chats is present, the virtual room selection will continue to use the Chats integration.
	•	If only WSC is present, the system will use the WSC integration for virtual room selection.
	•	If both Chats and WSC are present, the Chats integration will be used by default.

This change ensures proper handling of the WSC module integration where applicable, while maintaining backward compatibility with the Chats module.

Issue reference: https://zextras.atlassian.net/browse/WSC-1715

- [x] Test this code on a VM with both Chats and Workstream Collaboration

### CODE REVIEW

**Review by nubsthead — DISMISSED** [review]

**Review by CerDennis — DISMISSED** [review]

**Review by yukiLu6 — DISMISSED** [review]

File: src/view/editor/parts/editor-virtual-room.test.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+const FakeWscIntegrationComponent = jest.fn(
+	(): React.JSX.Element => <div data-testid={wscIntegrationDataTestId} />
+);
+
+describe('Editor virtual rooms', () => {
+	test('If only ChatsRoomSelector is available, it should be rendered', () => {
+		jest
+			.spyOn(shell, 'useIntegratedComponent')
+			.mockReturnValueOnce([FakeChatsIntegrationComponent, true]);
+		jest
```
Comment (gnekoz) [inline src/view/editor/parts/editor-virtual-room.test.tsx]: This mock approach relies heavily on the component current implementation. If the order of the calls to the `useIntegrationComponent` will change, the component will behave the same, but the tests will fail.
I suggest to use the `mockImplementation` method instead, and, based on the parameter, return a specific value

File: src/view/editor/parts/editor-virtual-room.test.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	test('If ChatsRoomSelector and WscRoomSelector are both available, only ChatsRoomSelector should be rendered', () => {
+		jest
+			.spyOn(shell, 'useIntegratedComponent')
+			.mockReturnValueOnce([FakeChatsIntegrationComponent, true]);
+		jest
+			.spyOn(shell, 'useIntegratedComponent')
+			.mockReturnValueOnce([FakeWscIntegrationComponent, true]);
+		setupTest(<EditorVirtualRoom editorId={'editorId'} />, { store });
+
+		expect(screen.getByTestId(chatsIntegrationDataTestId)).toBeInTheDocument();
```
Comment (gnekoz) [inline src/view/editor/parts/editor-virtual-room.test.tsx:57]: To be bulletproof it has to check also that the _wscIntegrationDataTestId_ is not in the document

File: src/view/editor/parts/editor-virtual-room.test.tsx (hunk 1 above)
Comment (lisaparma) [reply src/view/editor/parts/editor-virtual-room.test.tsx]: You're right! To do the mock implementation, I had to modify the initial declaration of the `useIntegratedComponent` mock by adding the `id` parameter, which I use to return the correct values. I opened a PR in the submodule https://github.com/zextras/carbonio-ui-commons/pull/152 to add it.

Comment (gnekoz) [reply src/view/editor/parts/editor-virtual-room.test.tsx]: Ouch! we forgot to change the type as we did for `useIntegratedFunction` and `useIntegratedAction`. Thanks for fixing it

**Review by gnekoz — DISMISSED** [review]

**Review by nubsthead — APPROVED** [review]

**Review by CerDennis — APPROVED** [review]

**Pull request merged**